nilearn/regions/region_extractor.py

```diff
@@ -0,0 +1,480 @@
+
+    return threshold_maps_img
+
+
+def break_connected_components(map_img, min_size=20,
+                               extract_type='connected_components',
+                               peak_local_smooth=6, mask_img=None):
+    """ A function takes the connected components of the brain activation
+    maps/regions and breaks each component into a seperate brain regions.
+
+    Parameters
+    ----------
+    map_img: a Nifti-like image/object
+        a 3D image of the activation maps which should be breaked into a set
+        of regions.
+
+    min_size: int, default is 20
+        An integer which denotes the size of voxels in the each region.
+        Only the size of the regions which are more than this number
+        are kept.
+
+    extract_type: string {"connected_components", "local_regions"} \
+        default is connected_components
+        A method used to segment/seperate the regions.
+
+        If connected_components, each component in the input image is assigned
+        a unique label point and then seperated based on the uniqueness of those
+        label points.
+
+        If local_regions, smoothing followed by random walker procedure is
+        used to split into each a seperate regions.
+
+    peak_local_smooth: scalar, default is 6mm
+        a value in mm which is used to smooth an image to locate seed points.
+
+    mask_img: Nifti-like image/object, default is None
+        an option used to mask the input brain map image.
+
+    Returns
+    -------
+    regions_accumulated: a Nifti-like images
+        contains the images of segmented regions each 3D image appended as a
+        seperate brain activated region.
+    """
+    regions_accumulated = []
+    map_img = check_niimg(map_img)
+    if len(map_img.shape) == 0 or len(map_img.shape) == 4:
+        raise ValueError('A 3D Nifti image or path to a 3D image should '
+                         'be submitted.')
+
+    extract_methods = ['connected_components', 'local_regions']
+    if extract_type not in extract_methods:
+        message = ("'extract_type' should be given "
+                   "either of these {0}").format(extract_methods)
+        raise ValueError(message)
+
+    map_data = map_img.get_data()
+    affine = map_img.get_affine()
+    # Mark the seeds using random walker
+    if extract_type == 'local_regions':
+        smooth_map_data = _smooth_array(map_data, affine, peak_local_smooth)
+        seeds = peak_local_max(smooth_map_data, indices=False,
+                               exclude_border=False)
+        seeds_label, seeds_id = label(seeds)
+        # Assign integer "-1" to ignore
+        seeds_label[map_data == 0.] = -1
+        # Take the maximum value of the data and
+        # normalize to be between -1 and 1
+        max_value_map = max(map_data.max(), -map_data.min())
+        if max_value_map > 1.:
+            map_data /= max_value_map
+        rw_maps = random_walker(map_data, seeds_label, mode='cg_mg')
```

Comment:
'cg_mg' may not work, you should do a try / except and downgrade to 'cg' then 'bf'.